cupyx/scipy/linalg/decomp_lu.py

```diff
@@ -71,8 +71,14 @@ def lu_factor(a, overwrite_a=False, check_finite=True):
     elif dtype.char == 'd':
         getrf = cusolver.dgetrf
         getrf_bufferSize = cusolver.dgetrf_bufferSize
+    elif dtype.char == 'F':
+        getrf = cusolver.cgetrf
+        getrf_bufferSize = cusolver.cgetrf_bufferSize
+    elif dtype.char == 'D':
+        getrf = cusolver.zgetrf
+        getrf_bufferSize = cusolver.zgetrf_bufferSize
     else:
-        raise NotImplementedError('Only float32 and float64 are supported.')
+        raise ValueError('unsupported dtype (actual:{})'.format(dtype))
```

Comment:
Could you keep `NotImplementedError` for now?  SciPy uses `float64` for `float16` or integer inputs.